### PR TITLE
Add the flink demo

### PR DESCRIPTION
**Pull request by yuanwu2017**
Signed-off-by: yuanwu <yuan.wu@intel.com>

### CODE REVIEW

File: demos/flink/run_flink_on_occlum_glibc.sh [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	-D taskmanager.memory.task.heap.size=402653174b \
+	-D taskmanager.memory.task.off-heap.size=0b &
+
+}
+
+run_task() {
+    export FLINK_CONF_DIR=/home/yuanwu/occlum/demos/flink/flink-1.10.1/conf && \
+        ./flink-1.10.1/bin/flink run ./flink-1.10.1/examples/streaming/WordCount.jar
+    #export FLINK_CONF_DIR=./flink-1.10.1/conf && \
+    #     ./flink-1.10.1/bin/flink run ./flink-1.10.1/examples/streaming/IncrementalLearning.jar  --input $2  --output $3
```
Comment (guzongmin) [inline demos/flink/run_flink_on_occlum_glibc.sh:89]: How to use those options? Can you please give some comments for them?

File: .github/workflows/demo_test.yml [hunk 2]
```diff
@@ -400,3 +400,34 @@ jobs:
...
+    - name: Download flink
+      run: docker exec flink_test bash -c "cd /root/occlum/demos/flink && ./download_flink.sh"
+
+    - name: Run jobmanager on host
+      run: docker exec flink_test bash -c "cd /root/occlum/demos/flink && SGX_MODE=SIM ./run_flink_webserver_on_host.sh"
+
+    - name: Run flink taskmanager
+      sleep ${{ env.nap_time }};
+      run: docker exec flink_test bash -c "cd /root/occlum/demos/flink && SGX_MODE=SIM ./run_flink_on_occlum_glibc.sh tm"
+    - name: Run flink task
```
Comment (guzongmin) [inline .github/workflows/demo_test.yml:432]: Please add new blank new line.

File: demos/flink/run_flink_on_occlum_glibc.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	-D taskmanager.memory.managed.size=536870920b \
+	-D taskmanager.cpu.cores=1.0 \
+	-D taskmanager.memory.task.heap.size=402653174b \
+	-D taskmanager.memory.task.off-heap.size=0b
+
+}
+
+run_task() {
+    
+    export FLINK_CONF_DIR=/home/yuanwu/occlum/demos/flink/flink-1.10.1/conf && \
```
Comment (guzongmin) [inline demos/flink/run_flink_on_occlum_glibc.sh:68]: Please update the path name.

File: demos/flink/run_flink_webserver_on_host.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64/
+workdir=$PWD
+echo "$workdir"
+java  -Xms1024m -Xmx1024m  -Dlog.file=$workdir/flink-1.10.1/log/flink--standalonesession-0-4bf84744d45d.log -Dlog4j.configuration=file:$workdir/flink-1.10.1/conf/log4j.properties -Dlogback.configurationFile=file:$workdir/flink-1.10.1/conf/logback.xml -classpath $workdir/flink-1.10.1/lib/flink-table-blink_2.11-1.10.1.jar:$workdir/flink-1.10.1/lib/flink-table_2.11-1.10.1.jar:$workdir/flink-1.10.1/lib/log4j-1.2.17.jar:$workdir/flink-1.10.1/lib/slf4j-log4j12-1.7.15.jar:$workdir/flink-1.10.1/lib/flink-dist_2.11-1.10.1.jar::: org.apache.flink.runtime.entrypoint.StandaloneSessionClusterEntrypoint --configDir $workdir/flink-1.10.1/conf --executionMode cluster &
```
Comment (guzongmin) [inline demos/flink/run_flink_webserver_on_host.sh:4]: The log name "flink--standalonesession-0-4bf84744d45d.log" could be more simple.

File: demos/flink/run_flink_on_occlum_glibc.sh [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        .env.default = [ "LD_LIBRARY_PATH=/usr/lib/jvm/java-11-openjdk-amd64/lib/server:/usr/lib/jvm/java-11-openjdk-amd64/lib:/usr/lib/jvm/java-11-openjdk-amd64/../lib:/lib" ]' Occlum.json)" && \
+    echo "${new_json}" > Occlum.json
+
+}
+
+build_flink() {
+    # Copy JVM and class file into Occlum instance and build
+    mkdir -p image/usr/lib/jvm
+    cp -r /usr/lib/jvm/java-11-openjdk-amd64 image/usr/lib/jvm
+    cp /lib/x86_64-linux-gnu/libz.so.1 image/lib
```
Comment (guzongmin) [inline demos/flink/run_flink_on_occlum_glibc.sh:33]: Try to use libz.so.1 in $occlum_glibc folder.

**Comment by yuanwu2017** [comment]
abandon

**Pull request closed without merging**